Simulator/step_function.py

```diff
@@ -573,84 +573,93 @@ def batch_shop(self, shop_action, player, game_observation):
+                case 1:
+                    # Buy from shop
+                    champ_shop_target =  np.argmax(action[6:11])
+                    self.batch_shop(champ_shop_target, player, game_observations[key])
+                case 2:
+                    # Swap champ place
+                    target_1 = np.argmax(action[6:44])
+                    action[target_1] = 0
+                    target_2 = np.argmax(action[6:45])
+                    swap_loc_from = min(target_1, target_2)
+                    swap_loc_to = max(target_1, target_2)
+                    if swap_loc_to == 36:
+                        # Sell Champ
+                        if swap_loc_from < 27:
+                            x,y = self.dcord_to_2dcord(swap_loc_from)
+                            if player.board[x][y]:
+                                player.sell_champion(player.board[x][y], field=True):
+                        else:
+                            player.sell_from_bench(swap_loc_from - 27)
+                    else:
+                        # Swap from swap_loc_from to swap_loc_to
+                        if swap_loc_from < 27:
+                            if swap_loc_to < 27:
+                                x1,y1 = self.dcord_to_2dcord(swap_loc_from)
+                                x2,y2 = self.dcord_to_2dcord(swap_loc_to)
+                                if player.board[x1][y1]:
+                                    player.move_board_to_board(x1, y1, x2, y2)
+                                elif player.board[x2][y2]:
+                                    player.move_board_to_board(x2, y2, x1, y1)
+                            else:
+                                x1,y1 = self.dcord_to_2dcord(swap_loc_from)
+                                bench_loc = swap_loc_to - 27
+                                if player.bench[bench_loc]:
+                                    player.move_bench_to_board(bench_loc, x1, y1)
+                                else:
+                                    player.move_board_to_bench(x1, y1)
+                case 3:
+                    # Place item on champ
+                    item_selector = np.argmax(action[45:55])
+                    move_loc = np.argmax(action[6:44])
+                    if move_loc >= 27:
+                        move_loc -= 27
+                        player.move_item_to_bench(item_selector, move_loc)
+                    else:
+                        x = move_loc % 7
+                        y = (move_loc - y ) // 7
+                        player.move_item_to_board(item_selector, x, y)
+                case 4:
+                    # Buy EXP
+                    player.buy_exp()
+                    pass
+                case 5:
+                    # Pass action
+                    pass
+            # game_observations[key].generate_game_comps_vector()
+            game_observations[key].generate_other_player_vectors(player, players)
+            return player.reward
+        return 0
```

Comment:
I'm not sure if this is actually needed. I'm pretty sure we get the reward in pettingzoo via the player directly and not via the step function so there is no need for a return.